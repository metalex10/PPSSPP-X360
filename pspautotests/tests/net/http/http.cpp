// WARNING: This test doesn't actually work yet. SendRequest comes back with some DNS error on the real PSP.


#include <common.h>
#include <pspnet.h>
#include <psphttp.h>
#include <psputility_netmodules.h>

extern "C" int main(int argc, char *argv[]) {
  int tmpl, connid, reqid;

	checkpointNext("Init:");
	checkpoint("sceUtilityLoadNetModule common: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_COMMON));
	checkpoint("sceUtilityLoadNetModule adhoc: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_ADHOC));
	checkpoint("sceUtilityLoadNetModule inet: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_INET));
	checkpoint("sceUtilityLoadNetModule parsehttp: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_PARSEHTTP));
	checkpoint("sceUtilityLoadNetModule parseuri: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_PARSEURI));
	checkpoint("sceUtilityLoadNetModule http: %08x", sceUtilityLoadNetModule(PSP_NET_MODULE_HTTP));
	checkpoint("sceNetInit: %08x", sceNetInit(65536, 30, 0x1000, 30, 0x1000));
	checkpoint("sceHttpInit: %08x", sceHttpInit(20000)); 
//	checkpoint("sceHttpInitCache: %08x", sceHttpInitCache(0x1000)); 

  checkpoint("sceHttpCreateTemplate: %08x", tmpl = sceHttpCreateTemplate("pspautotests", 1, 0));  // 1, 0 = unknown but works

  checkpointNext("Connecting to ppsspp.org:");
  checkpoint("sceHttpCreateConnectionWithURL: %08x", connid = sceHttpCreateConnectionWithURL(tmpl, "http://www.ppsspp.org", 0));
  checkpoint("sceHttpCreateRequestWithURL: %08x", reqid = sceHttpCreateRequestWithURL(connid, PSP_HTTP_METHOD_GET, "/test.txt", 0));
  checkpoint("sceHttpSendRequest: %08x", sceHttpSendRequest(reqid, 0, 0));

  SceULong64 contentLength;
  checkpoint("sceHttpGetContentLength: %08x", sceHttpGetContentLength(reqid, &contentLength));
  checkpoint("response length: %08x", (int)contentLength);
  
  char *buffer = new char[contentLength + 1];
  buffer[contentLength] = 0;
  checkpoint("sceHttpReadData: %08x", reqid, buffer, contentLength);
  checkpoint("content: %s", buffer);
  delete [] buffer;

	checkpointNext("Cleanup:");
  checkpoint("sceHttpDeleteRequest: %08x", sceHttpDeleteRequest(reqid));
	checkpoint("sceHttpDeleteConnection: %08x", sceHttpDeleteConnection(connid));
	checkpoint("sceHttpDeleteTemplate: %08x", sceHttpDeleteTemplate(tmpl));
	checkpoint("sceHttpEnd: %08x", sceHttpEnd());
  checkpoint("sceNetTerm: %08x", sceNetTerm());
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_HTTP));
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_PARSEURI));
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_PARSEHTTP));
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_INET));
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_ADHOC));
	checkpoint("sceUtilityUnloadModule: %08x", sceUtilityUnloadNetModule(PSP_NET_MODULE_COMMON));
  // etc...
	return 0;
}
