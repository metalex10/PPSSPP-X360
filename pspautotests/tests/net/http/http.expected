[x] Init:
[r] sceUtilityLoadNetModule common: 00000000
[r] sceUtilityLoadNetModule adhoc: 00000000
[r] sceUtilityLoadNetModule inet: 00000000
[r] sceUtilityLoadNetModule parsehttp: 00000000
[r] sceUtilityLoadNetModule parseuri: 00000000
[r] sceUtilityLoadNetModule http: 00000000
[x] sceNetInit: 00000000
[x] sceHttpInit: 00000000
[x] sceHttpCreateTemplate: 00000001

[x] Connecting to ppsspp.org:
[x] sceHttpCreateConnectionWithURL: 01000001
[x] sceHttpCreateRequestWithURL: 02000001
[x] sceHttpSendRequest: 80410402
[x] sceHttpGetContentLength: 80431065
[x] response length: ffffffff
[x] sceHttpReadData: 02000001
[x] content: 

[x] Cleanup:
[x] sceHttpDeleteRequest: 00000000
[x] sceHttpDeleteConnection: 00000000
[x] sceHttpDeleteTemplate: 00000000
[x] sceHttpEnd: 00000000
[x] sceNetTerm: 00000000
[r] sceUtilityUnloadModule: 00000000
[r] sceUtilityUnloadModule: 00000000
[r] sceUtilityUnloadModule: 00000000
[r] sceUtilityUnloadModule: 00000000
[r] sceUtilityUnloadModule: 00000000
[r] sceUtilityUnloadModule: 00000000
